app/src/main/kotlin/compiler/Compiler.kt

```diff
@@ -29,9 +26,6 @@ class Compiler(val diagnostics: Diagnostics) {
         val parseTree = parser.process(leaves)
 
         val ast = AstFactory.createFromParseTree(parseTree, diagnostics)
-        val nameResolution = NameResolver.calculateNameResolution(ast, diagnostics)
-        val argumentResolution = ArgumentResolver.calculateArgumentToParameterResolution(ast, nameResolution, diagnostics)
-        val expressionTypes = TypeChecker.calculateTypes(ast, nameResolution, argumentResolution, diagnostics)
-        val variableProperties = VariablePropertiesAnalyzer.calculateVariableProperties(ast, nameResolution, diagnostics)
+        val programProperties = Resolver.resolveProgram(ast, diagnostics)
```

Comment:
I refactored this a little bit to hide some details, let me know what do you think.